### ISSUE TITLE
Basic Work? (Finding new textures)

### ISSUE THREAD

**Issue by JLPrescott** [traverseda/PyCraft#60]
Hi All,

I am offering my assistance here towards this project as for a class assignment I have been tasked to contribute to an open-sourced project like this one. I have spoken with @traverseda about this and he said it would be a good idea to open an issue.

I have a short period of time to complete this (less than 4 weeks) and unfortunately I have no experience with Python coding and don't think I will have the time to learn it fluently enough to contribute in that area. I'm looking for any suggestions of basic and non-time-demanding tasks that would help this project and fulfil my requirement to contribute.

Any suggestions will be appreciated here.

Regards,
James

**Comment by traverseda** [comment]
I'd like to find some permissibly licensed art assets. You could start on opengameart.

What I'd like is a high-quality minecraft texture pack, and some decent character models. For the character model, I'd like something more toon-ish then blocky.

The license is really what matters there. You can find a bit more about licensing [here](http://www.gnu.org/licenses/license-list.en.html).

**Comment by gbear605** [comment]
See https://github.com/SirCmpwn/TrueCraft/issues/45 for free (as in freedom) resources used by another minecraft clone project.

**Comment by TristanTrim** [comment]
@JLPrescott
It's kinda hard to give advice on this without knowing more context on your skills, assignment, and etc.
You could try helping out on #34 with clarifying goals. Possibly trying to lead discussion and collect opinions. You could poll some communities that aren't dev's here for things they'd like to see in an open source minecraft clone.

**Comment by JLPrescott** [comment]
Hi guys,

I will chase up some permissibly licensed art assets. I notice that you only have implemented grass, sand, brick and stone blocks into the game. What is the extent of textures do you want me to find. Will there be new blocks that haven’t been added just yet that I should be adding to the textures? I'm assuming that you don't want too high of a resolution as it may make the game struggle. Does anyone have a preference for resolution? 

All my assignment requires me to do is to contribute any amount to an open source project and present the work I have done to the class. I think a visual contribution such as texture updates would be the best for me to show to a group of people.

**Comment by traverseda** [comment]
We'd like to keep our options open, I think. A good wide selection would be great.

**Comment by JLPrescott** [comment]
Hi all,
So I have been looking around for block textures and so far have come up with a few to work with. I just want to see whether this is the type of textures you were after. These are all licensed under CC0.
![textures](https://cloud.githubusercontent.com/assets/18594743/14940643/1accde76-0fc2-11e6-89ab-f175dbb9bcc9.PNG)

@Gbear605 also pointed out that there is some voxel textures designed around minecraft that is available for any free use, however I don't think this is the sort of low quality textures you were after. See: https://github.com/deathcap/ProgrammerArt 

For the most part here I just want to make sure I'm on the right track and in line with what you wanted. Your feedback would be great.

**Comment by traverseda** [comment]
It's going in a pretty realistic direction, and I for one would prefer something more toon-ey.

See [graphics vs aesthetics](https://www.youtube.com/watch?v=5oK8UTRgvJU).

But functionally, those textures look like they'd work.

I'd recommend looking at some existing minecraft themes that look good and, if you can't find a good looking one that's permissibly licensed, contacting the creator and asking them to license it under something like the creative-commons.

**Comment by JLPrescott** [comment]
Yeah ok. You want something more like this texture pack: http://bdcraft.net/
I'll see what I can come up with. Unfortunately trying to find stuff that is toony and also something where we don't have to recognise others with can be a bit tricky. I'll work on it.

**Comment by traverseda** [comment]
I don't have any particular problem with attribution. I would prefer a license that allowed derivatives and commercial activity though. We might want to start a patreon or otherwise raise funds for this some time in the distant future.

**Comment by JLPrescott** [comment]
Hi again,
I did a little looking around under the 'toon' styling today. What do you think of the following:
![capture](https://cloud.githubusercontent.com/assets/18594743/14953961/c813ade0-10b1-11e6-8610-d965f37c405b.PNG)
This is harder than it seems as I am trying to cater for a wide range of people, keep within the toon styling and also find free to use textures.
If you aren't into what i've found so far we could use something like this texture pack: http://piehole.alexvoelk.de/ 
It's licensed under CC and only have to worry about attributions.

**Comment by traverseda** [comment]
I have to admit, I find the ones you linked to pretty ugly, personally. But the piehole texture pack looks pretty good, and I've got no problem with including attribution.

If no one else speaks up, I'm inclined to use piehole.

**Comment by JLPrescott** [comment]
Yeah I agree. I tried haha

So I managed to download and overwrite the pre-existing textures.png file with the same textures from the Piehole texture pack. I've tested it and it works fine. I guess I'll create a pull request shortly to reflect this. Before I do this, should I include my attribution to the texture pack author in the license file, or would you prefer it elsewhere?

**Comment by traverseda** [comment]
I think the right place for it, for now, if in the readme. We'll put it in the menu when we have a menu.

**Comment by JLPrescott** [comment]
Okay so now I'm getting somewhere!

I have a texture file with all the blocks from the original minecraft, but I think there needs to be a major rework of how the application pulls in each texture before adding new blocks. I made that temporary texture pack by replacing each previous texture space, but this can't really be moved further until more blocks are added into the game.

I also tried to take a screenshot to replace the current one now that the textures are different, but my copy of pycraft was only generating really unimpressive slabs of terrain that varied maybe 3 or 4 blocks in height for some reason. There wasn't any definitive hills like the current image shows, so I just left that for now.
